src/providers/weights.rs

```diff
@@ -62,17 +62,27 @@ const PERFECT_RATIO: f64 = 1.0;
 
 fn calculate_chain_weight(
     provider_availability: Availability,
-    chain_availability: &Availability,
-) -> u32 {
+    chain_availability: Availability,
+) -> u64 {
     let Availability(provider_success, provider_failure) = provider_availability;
 
     // Sum failed and successful calls for provider
-    let provider_total = provider_success + provider_failure * provider_failure;
+    let Some(provider_failures_squared) = provider_failure.checked_mul(provider_failure) else {
+        // 1 is minimal value for chein weight
```

Comment:
Do we want a minimal value or a maximal value? Min would result in thrashing, no?
```suggestion
        // 1 is minimal value for chain weight
```

Do we really expect this to overflow given the increase to u64? Is there a better way to handle very large numbers?

Reply:
Not sure what would be the best way to handle this, this is mainly a hotfix, as I want to get it back online.
https://github.com/WalletConnect/rpc-proxy/issues/232
here's the task for future improvements

Reply:
Actually, defaulting to 1 will result in over-prioritizing this provider, no? So 1 is definitely worse than defaulting to `provider_failure`?

Reply:
It's also possible we hit an overflow down below:
```rust
let provider_total = provider_success + provider_failures_squared;
```

Reply:
But this 1 is not a value we give to provider_squared. It's value of weight (minial possible weight) we give to this

Reply:
Hence the return function

Reply:
What happens if we set the weight of a provider to 0? We should do that if availability goes below e.g. 90%.

Reply:
I'd return to the edge case on addition with the other issue, for now just want to get it back online, and this should work in 99,9% of cases

Reply:
If we set it to 0 we will disable the provider. And actually I think this is what we should do